submissions/antonkottans/tjsw-post-oop/index.js

```diff
@@ -0,0 +1,185 @@
+const KEYS_VALUE_SEPARATOR = ": ";
+const PROPS_SEPARATOR = "; ";
+
+class Creature {
+    constructor({ name, gender, friends = [], species, saying }) {
+        this.name = name;
+        this.species = species;
+        this.gender = gender;
+        this.friends = friends;
+        this.saying = saying;
+    }
+
+    createArrayOfProps() {
+        return ["name", "species", "gender", "saying", "friends"]
+            .map((key) => {
+                if (key != "friends")
+                    return key + KEYS_VALUE_SEPARATOR + this[key];
+                return this.friends.length
+                    ? "friends: " +
+                          this.friends.map((friend) => friend.name).join(", ")
+                    : "";
+            })
+            .filter((_) => _);
+    }
+
+    addFriends(...newFriends) {
+        if (newFriends.length > 0)
+            this.friends = [...this.friends, ...newFriends];
```

Comment:
If we do not check newFriends length how it would affect outcome from `[...this.friends, ...newFriends]`?